stl/inc/chrono

```diff
@@ -6202,6 +6163,128 @@ namespace chrono {
             }
         }
 
+        template <class _Ty>
+        void _Validate_specifiers(const _Chrono_spec<_CharT>& _Spec, const _Ty& _Val) {
+            // clang-format off
+            if constexpr (
+                _Is_specialization_v<_Ty, duration> || is_same_v<_Ty, sys_info>
+                || _Is_specialization_v<_Ty, time_point> || _Is_specialization_v<_Ty, _Local_time_format_t>) {
+                return;
+            }
+            // clang-format on
+
+            if constexpr (_Is_specialization_v<_Ty, hh_mm_ss>) {
+                if (_Spec._Type == 'H' && _Val.hours() >= hours{24}) {
+                    _THROW(format_error("Cannot localize hh_mm_ss with an absolute value of 24 hours or more."));
+                }
+                return;
+            }
+
+            constexpr bool _Is_ymd =
+                _Is_any_of_v<_Ty, year_month_day, year_month_day_last, year_month_weekday, year_month_weekday_last>;
+
+            const auto _Validate = [&] {
+                switch (_Spec._Type) {
+                case 'a':
+                case 'A':
+                case 'u':
+                case 'w':
+                    if constexpr (_Is_any_of_v<_Ty, weekday, weekday_last>) {
+                        return _Val.ok();
+                    } else if constexpr (_Is_any_of_v<_Ty, weekday_indexed, year_month_weekday,
+                                             year_month_weekday_last>) {
+                        return _Val.weekday().ok();
+                    } else if constexpr (is_same_v<_Ty, month_weekday>) {
+                        return _Val.weekday_indexed().weekday().ok();
+                    } else if constexpr (is_same_v<_Ty, month_weekday_last>) {
+                        return _Val.weekday_last().ok();
+                    } else if constexpr (_Is_any_of_v<_Ty, year_month_day, year_month_day_last>) {
+                        return _Val.ok();
+                    }

```

Comment:
Should we still format the weekday if just the `year`/`month` part of these composite types is `!ok()`?  All of the other checks are zeroing in on the `ok()`-ness of the weekday component alone, whereas this check seems to require that the entire type is `ok()`.

Reply:
For these types, I don't believe that's possible. They don't directly contain `weekday` information - it has to be calculated from the time point being represented. Given `2021y / month{15} / 7` or `2021y / month{15} / last`, it's impossible to know what `weekday` that would be. (For a bogus `year` only, we might theoretically be able to calculate what `year{1'000'000} / January / 1` would be, but it's probably easier if we don't have to handle extreme inputs in the weekday-calculating algorithm.)

Reply:
Ah good call -- thanks!  I missed that piece of logic ;)

Reply:
The funny thing is, there is only one bogus year, and it's `-32768`.